File: testers/avro/java-spring/generated-and-checked-in/com/example/events/common/MoneyPublisher.java
package com.example.events.common;

import java.util.concurrent.CompletableFuture;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.kafka.support.SendResult;
import org.springframework.stereotype.Service;

/** Type-safe event publisher for money topic */
@Service
public record MoneyPublisher(KafkaTemplate<String, Money> kafkaTemplate, String topic) {
  public MoneyPublisher(KafkaTemplate<String, Money> kafkaTemplate) {
    this(kafkaTemplate, "money");
  }

  public MoneyPublisher withKafkaTemplate(KafkaTemplate<String, Money> kafkaTemplate) {
    return new MoneyPublisher(kafkaTemplate, topic);
  }

  public MoneyPublisher withTopic(String topic) {
    return new MoneyPublisher(kafkaTemplate, topic);
  }

  /** Publish a Money event */
  public CompletableFuture<SendResult<String, Money>> publish(String key, Money event) {
    return kafkaTemplate.send(topic, key, event);
  }
}

File: testers/avro/scala-cats/generated-and-checked-in/com/example/events/common/MoneyPublisher.scala
package com.example.events.common

import cats.effect.IO
import fs2.kafka.KafkaProducer
import fs2.kafka.ProducerRecord
import fs2.kafka.ProducerRecords
import fs2.kafka.ProducerResult
import scala.annotation.unused

@unused
/** Type-safe event publisher for money topic */
case class MoneyPublisher(
  kafkaTemplate: KafkaProducer[IO, String, Money],
  topic: String = "money"
) {
  /** Publish a Money event */
  def publish(
    key: String,
    event: Money
  ): IO[ProducerResult[String, Money]] = {
    return kafkaTemplate.produce(ProducerRecords.one(ProducerRecord(topic, key, event))).flatten
  }
}